### ISSUE TITLE
b2c - CryptographicException: Bad Length. - error

### ISSUE THREAD

**Issue by vmureithi** [mutiadavid/MpesaSDK.NET#9]
string credential = "pass".ToMpesaSecurityCredential();

var result = await mpesaclient.B2CAsync("initiatorUsername", credential,…….);

How do we rectify the problem?

regards,

Vincent

**Comment by mutiadavid** [comment]
Hey, could you include the exception you are getting please, sharing ex.ToString() would help me to know why we might be getting the exception.

**Comment by mutiadavid** [comment]
Actually you don't need to call pass.ToMpesaSecurityCredential(), this will be called internally.  B2CAsync accepts the plain password then handles the encryption for you.

**Comment by vmureithi** [comment]
Yes I realized I was passing the encrypted password, and saw from your source code that you are already doing it.

Secondly, I just realized I needed to catch up with the grueling, brutal, ruthless in regards to the customer experience of mpesa API integration. The process, terminologies etc. It has been awful I have to say.

Finally, I did manage to do a STKPushAsync which is working perfectly.

Kind Regards,

Vincent.
